### DISCUSSION TITLE
Creation of the exchanges, queues, and their bindings

### DISCUSSION THREAD

**Discussion by kstojanovski-work** [spring-projects/spring-amqp#2470] [Q&A] [answered]
Should the exchanges, queues, and bindings need to be built by Spring AMQP or manually on the RabittMQ?
What is the best practice?

**Comment by artembilan** [comment, marked as answer]
It is very convenient on the development phase, that the project manages those RabbitMQ object.
In production I'd recommend to manage them on broker side, when some permissions and other restrictions may apply.
If they exist on a broker, Spring AMQP does not override them.
This way the code might remain in your project and it won't hurt in production.
Unless some mismatched basic properties are different.

In the end both variants are OK: might fully depend on the development team preferences.
For example Spring Cloud Stream Binder abstraction even has a dedicated feature to populate (or modify) those broker objects for bindings.
